src/sbl_filing_api/routers/filing.py

```diff
@@ -59,6 +59,34 @@ async def post_filing(request: Request, lei: str, period_code: str):
         )
 
 
+@router.put("/institutions/{lei}/filings/{period_code}/sign", response_model=FilingDTO)
+@requires("authenticated")
+async def sign_filing(request: Request, lei: str, period_code: str):
+    filing = await repo.get_filing(request.state.db_session, lei, period_code)
+    if not filing:
+        return JSONResponse(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            content=f"There is no Filing for LEI {lei} in period {period_code}, unable to sign a non-existent Filing.",
+        )
+    latest_sub = await repo.get_latest_submission(request.state.db_session, lei, period_code)
+    if not latest_sub or latest_sub.state != SubmissionState.SUBMISSION_ACCEPTED:
+        return JSONResponse(
+            status_code=status.HTTP_403_FORBIDDEN,
+            content=f"Cannot sign filing. Filing for {lei} for period {period_code} does not have a latest submission the SUBMISSION_ACCEPTED state.",
+        )
+    if not filing.contact_info:
+        return JSONResponse(
+            status_code=status.HTTP_403_FORBIDDEN,
+            content=f"Cannot sign filing. Filing for {lei} for period {period_code} does not have contact info defined.",
+        )
+    sig = await repo.add_signature(
+        request.state.db_session, signer_id=request.user.id, signer_name=request.user.name, filing_id=filing.id
+    )
+    filing.confirmation_id = lei + "-" + period_code + "-" + str(latest_sub.id) + "-" + str(sig.signed_date.timestamp())
+    filing.signatures.append(sig)
```

Comment:
How do we know when a filing needs to be re-signed once changes have been made to an already-signed filing?

We _could_...

1. Associate signatures to submissions rather than filings. Then on the signing screen if the submission associated to the filing isn't signed, you now you need to sign it? That wouldn't fix needing to re-sign after changing contact info or institution data, though.
2. Disassociate a signature from a filing if anything changes on a signed filing. Not sure how that'd work in the current one-to-many sigs per filing. Add a `invalid`/`expired` type flag?

    Alternatively, don't have it as a one-to-many, and just hang the signature off the filing record similar to other fields. If anything changes on the filing, the signature gets dropped. We lose history of signature, but makes it simple. If we want history, we could implement it the same way we do for institution data at a later time.

Anyway, I don't think we need to solve this for _this_ PR, but I think we will before we go live.

Reply:
Created https://github.com/cfpb/sbl-filing-api/issues/146